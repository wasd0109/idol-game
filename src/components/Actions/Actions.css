#performed {
  min-height: 50vh;
  max-height: 50vh;
}
