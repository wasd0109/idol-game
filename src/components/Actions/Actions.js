import React from 'react';
import './Actions.css';
import { connect } from 'react-redux';

import { performAction } from '../../actions';

const mapStateToProps = (state) => {
  console.log(state.receiveActionResults.actionResults);
  return {
    actionResults: state.receiveActionResults.actionResults,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    performAction: (event) => dispatch(performAction()),
  };
};

function Actions(props) {
  const { actionResults, performAction } = props;
  return (
    <div className="w-7/12 m-auto">
      <div id="performed" className="border-2 p-2 overflow-y-scroll">
        {actionResults.map((action, i) => {
          return (
            <h1 key={i} className="">
              {action}
            </h1>
          );
        })}
      </div>
      <div id="actions" className="flex justify-evenly my-2">
        <button
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1  px-2 rounded w-2/12"
          id="practice"
          onClick={performAction}
        >
          Practice
        </button>
        <button
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-2 rounded w-2/12"
          id="live"
        >
          Live
        </button>
        <button
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1  px-2 rounded w-2/12"
          id="tsunagari"
        >
          Tsunagari
        </button>
        <button
          className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-1  px-2 rounded w-2/12"
          id="tweet"
        >
          Tweet
        </button>
      </div>
    </div>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(Actions);
