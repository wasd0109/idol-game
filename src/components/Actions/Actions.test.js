import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Actions from './Actions';

let component;
const initialProps = {
  actionResults: [],
  userID: '0',
  performAction: jest.fn(),
  actions: ['Practice', 'Live', 'Tsunagari', 'Tweet'],
};

beforeEach(() => {
  component = render(<Actions {...initialProps} />);
});

const actions = ['Practice', 'Live', 'Tsunagari', 'Tweet'];

describe('Action component render as expected', () => {
  test('render Action component', () => {
    expect(component).toMatchSnapshot();
  });
  test('Render action results as expected', () => {
    const props = {
      actionResults: ['Test 1', 'Test 2', 'Test 3'],
      userID: '0',
      performAction: jest.fn(),
    };
    const component = render(<Actions {...props} actions={actions} />);
    expect(component.getByText('Test 1')).toBeInTheDocument();
    expect(component.getByText('Test 2')).toBeInTheDocument();
    expect(component.getByText('Test 3')).toBeInTheDocument();
  });
});

describe('Action component function as expected', () => {
  test('Pressing action button results in action', () => {
    const actionType = 'Practice';
    const actionBtn = screen.getByText(actionType);
    fireEvent.click(actionBtn);
    expect(initialProps.performAction).toBeCalled();
  });
  test.each(actions)('Action to be called with %s type', (action) => {
    const actionBtn = screen.getByText(action);
    fireEvent.click(actionBtn);
    expect(initialProps.performAction).toBeCalledWith(
      initialProps.userID,
    );
  });
  test('Triggering action results in cool down', () => {
    const actionType = 'Practice';
    const actionBtn = screen.getByText(actionType);
    fireEvent.click(actionBtn);
    expect(screen.getByText(/Cooling down/i));

    expect(screen.getByText(actionType).parentElement).toHaveAttribute('disabled');
  });
  test('Cool down end after timer run out', () => {
    const actionType = 'Practice';
    const actionBtn = screen.getByText(actionType);
    fireEvent.click(actionBtn);
    screen.findByText('Press button to perform action');
  });
});
