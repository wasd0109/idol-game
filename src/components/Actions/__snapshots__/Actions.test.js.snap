// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action component render as expected render Action component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border-2 p-2 overflow-y-scroll capitalize"
        id="performed"
      />
      <div
        class="flex justify-evenly my-2 flex-wrap md:no-wrap"
        id="actions"
      >
        <button
          class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
          id="Practice"
        >
          <p
            class="text-white font-bold"
          >
            Practice
          </p>
        </button>
        <button
          class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
          id="Live"
        >
          <p
            class="text-white font-bold"
          >
            Live
          </p>
        </button>
        <button
          class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
          id="Tsunagari"
        >
          <p
            class="text-white font-bold"
          >
            Tsunagari
          </p>
        </button>
        <button
          class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
          id="Tweet"
        >
          <p
            class="text-white font-bold"
          >
            Tweet
          </p>
        </button>
      </div>
      <div
        class="items-center bg-blue-500 text-white font-semibold px-4 py-3 rounded"
        role="alert"
      >
        Press button to perform action
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border-2 p-2 overflow-y-scroll capitalize"
      id="performed"
    />
    <div
      class="flex justify-evenly my-2 flex-wrap md:no-wrap"
      id="actions"
    >
      <button
        class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
        id="Practice"
      >
        <p
          class="text-white font-bold"
        >
          Practice
        </p>
      </button>
      <button
        class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
        id="Live"
      >
        <p
          class="text-white font-bold"
        >
          Live
        </p>
      </button>
      <button
        class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
        id="Tsunagari"
      >
        <p
          class="text-white font-bold"
        >
          Tsunagari
        </p>
      </button>
      <button
        class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
        id="Tweet"
      >
        <p
          class="text-white font-bold"
        >
          Tweet
        </p>
      </button>
    </div>
    <div
      class="items-center bg-blue-500 text-white font-semibold px-4 py-3 rounded"
      role="alert"
    >
      Press button to perform action
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
